com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DeleteServicesStep.java

```diff
@@ -96,63 +97,53 @@ private void validateGuidMapping(Map<String, String> serviceGuids) {
         CloudControllerClient client = execution.getControllerClient();
 
         Map<String, String> serviceNameGuidMap = services.parallelStream()
-            .collect(Collectors.toMap(Function.identity(), name -> {
-
-                Map<String, Object> serviceInstanceWithMetadata = serviceInstanceGetter.getServiceInstanceWithMetadata(client, name,
-                    spaceId);
-
-                @SuppressWarnings("unchecked")
-                String guid = (String) Stream.of(serviceInstanceWithMetadata)
-                    .filter(Objects::nonNull)
-                    .map(map -> (Map<String, Object>) map.get("metadata"))
-                    .filter(Objects::nonNull)
-                    .map(map -> map.get("guid"))
-                    .filter(Objects::nonNull)
-                    .findFirst()
-                    .orElse("");
-
-                return guid;
-
-            }));
+            .collect(Collectors.toMap(Function.identity(), name -> this.serviceNameToGuid(name, client, spaceId)));

```

Comment:
Could we do:
1. Find all the serviceWithMetadata for each service
2. Filter the ones which have empty guid
3. map them to the corresponding service?